File: src/app/_models/subject.ts
import { DbItem } from './db-item';

export class Subject extends DbItem {
  userId: string;
  courseId: string;
  name: string;
  items: Array<any>;

  constructor() {
    super();
  }

  getNotes(): Array<any> {
    if (this.items) {
      return this.items.filter(e => e.type == 'note');
    }

    return new Array<any>();
  }

  getQuestions(): Array<any> {
    if (this.items) {
      return this.items.filter(e => e.type === 'question');
    }

    return new Array<any>();
  }
}

File: lib/models/subject.dart
import 'package:studio/globals.dart';
import 'package:studio/models/question.dart';
import 'package:studio/models/subject_item.dart';

import 'note.dart';

class Subject {
  String id;
  String userId;
  String courseId;
  String name;
  String nameDb;
  DateTime created;
  List<Note> notes;
  String bookId;
  String bookTitle;
  List<Question> questions;
}